walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -22,73 +22,73 @@
-                            openTimeRanges,
-                            TimeRange.START_OF_DAY,
-                            eventTimeRanges.get(i).start(),
-                            false
-                        );
-                            if (eventTimeRanges.size() > 1) {
-                                addTimeRange(
-                                    openTimeRanges,
-                                    eventTimeRanges.get(i).end(),
-                                    eventTimeRanges.get(i+1).start(),
-                                    false
-                                );
-                            } else {
-                                addTimeRange(
-                                    openTimeRanges,
-                                    eventTimeRanges.get(i).end(),
-                                    TimeRange.END_OF_DAY,
-                                    true
-                                );
-                            }
-                    } else if (i == eventTimeRanges.size()-1) {
-                        addTimeRange(
-                            openTimeRanges,
-                            eventTimeRanges.get(i).end(),
-                            TimeRange.END_OF_DAY,
-                            true
-                        );
-                    } else {
-                        addTimeRange(
-                            openTimeRanges,
-                            eventTimeRanges.get(i).end(),
-                            eventTimeRanges.get(i+1).start(),
-                            false
-                        );
-                    }
+        for (int i = 0; i < eventTimeRanges.size(); i++) {
+            if (i == 0) {
+                addTimeRangeEndExcluded(
+                    openTimeRanges,
+                    TimeRange.START_OF_DAY,
+                    eventTimeRanges.get(i).start()
+                );
+                if (eventTimeRanges.size() > 1) {
+                    addTimeRangeEndExcluded(
+                        openTimeRanges,
+                        eventTimeRanges.get(i).end(),
+                        eventTimeRanges.get(i + 1).start()
+                    );
+                } else {
+                    addTimeRangeEndIncluded(
+                        openTimeRanges,
+                        eventTimeRanges.get(i).end(),
+                        TimeRange.END_OF_DAY
+                    );
                 }
-                return openTimeRanges;
-
+            } else if (i == eventTimeRanges.size() - 1) {
+                addTimeRangeEndIncluded(
+                    openTimeRanges,
+                    eventTimeRanges.get(i).end(),
+                    TimeRange.END_OF_DAY
+                );
             } else {
-                return Arrays.asList();
+                addTimeRangeEndExcluded(
+                    openTimeRanges,
+                    eventTimeRanges.get(i).end(),
+                    eventTimeRanges.get(i + 1).start()
+                );
```

Comment:
There's a lot of logic duplication here that can be streamlined to make this more straightforward. 
What you have is basically this: 
for items:
    if(isFirstEventOfDay) {
        addStartingFreeBlock
        if(moreEventsToday) {
             handleCommonCase
        } else { 
          addEndOfDayFreeBlock
        }
     } else if (isLastEventOfDay) {
        addEndOfDayFreeBlock
     } else {
          handleCommonCase
     }
     }

This can be simplified a lot to something like: 
for items:
     if(first event of day) {
          addInitialOpenBlock
      }
      if(last event of day) {
          addEndOfdayOpenBlock
      }
      if(more events) {
         handleCommonCase
       }
}

You could even move the special handling for first and last events outside of the loop if you wanted to. 
More conceptually, does this work for all situations? This seems to be assuming that after sorting the events will be non-overlapping. Will that be addressed in further commits? Are you adding tests for those cases?